2-mandatory/4-stocks.js

```diff
@@ -26,15 +22,20 @@ const CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS = [
             For example, the first element of the resulting array should contain Apple’s (aapl) average stock price for the last 5 days.
             The second element should be Microsoft's (msft) average price, and so on.
     The average value should be rounded to 2 decimal places, and should be a number (not a string)
-
     Hint 1: To calculate the average of a set of values, you can add them together and divide by the number of values.
         So the average of 5, 10 and 20 is (5 + 10 + 20) / 3 = 11.67
     Hint 2: If the problem seems complex, try breaking it down into smaller problems.
         Solve the smaller problems, and then build those solutions back up to solve the larger problem.
         Functions can help with this!
 */
-function getAveragePrices(closingPricesForAllStocks) {
-    // TODO
+//https://jrsinclair.com/articles/2019/five-ways-to-average-with-js-reduce/
+function getAveragePrices(closingPricesForAllStocks)
+{
+    //const reduce  = r => i => a => a.reduce(r, i);
+    //conatiner   =     getting items                      => got items.reduce          =>      add    /    length         .map previous  =>       round to 2 dec
+    const average = (closingPricesForAllStocks.map((array) => array.reduce((r, current) => r + current / array.length, 0))).map((element) => parseFloat(element.toFixed(2)));
```

Comment:
Great job,  just think about how you can break this down into smaller problems and functions.